### PR TITLE
Disable registration on unexpected origins

### PR DESCRIPTION
**Pull request by frederikrothenberger**
Disable registrations on unexpected origins, most notably https://identity.raw.ic0.app.
Using unexpected origins is a security risk and breaks the integration with dApps authenticating via Internet Identity.

### CODE REVIEW

**Comment by frederikrothenberger** [comment]
Screenshot of info message:
Screenshot 2021-12-15 at 14 59 03
The 'here' link points to the deprecation announcement here: https://forum.dfinity.org/t/internet-identity-deprecation-of-account-creation-on-all-origins-other-than-https-identity-ic0-app/9694

File: src/frontend/src/flows/loginUnknown.ts [hunk 1]
```diff
@@ -99,6 +100,14 @@ export const loginUnknown = async (
   });
 };
 
+function isRegistrationAllowed() {
```
Comment (nmattia) [inline src/frontend/src/flows/loginUnknown.ts:103]: a note here and some comments in `registerDisabled.ts` would be great!

File: src/frontend/src/flows/loginUnknown.ts [hunk 2]
```diff
@@ -99,6 +100,14 @@ export const loginUnknown = async (
...
+  return (
+    /^(https:\/\/identity\.ic0\.app$)|(https:\/\/[a-zA-Z0-9-]+\.[a-z]+\.dfinity\.network$)/.test(
```
Comment (nmattia) [inline src/frontend/src/flows/loginUnknown.ts:105]: There is exactly one endpoint (namely `identity.raw.ic0.app`) that you want to disable registration on. Why not simply check for that?

**Review by dubovitskaya — APPROVED** [review]
Thanks Frederik! The text looks good!
Just a small question - maybe we dont need the Logout button in this case?

File: src/frontend/src/flows/loginUnknown.ts (hunk 2 above)
Comment (frederikrothenberger) [reply src/frontend/src/flows/loginUnknown.ts:105]: Done. AFAIK we have to check against 3 origins:
https://identity.raw.ic0.app, https://rdmx6-jaaaa-aaaaa-aaadq-cai.ic0.app, https://rdmx6-jaaaa-aaaaa-aaadq-cai.raw.ic0.app

Are you sure there are not more?

Comment (nmattia) [reply src/frontend/src/flows/loginUnknown.ts:105]: I am not! I actually didn't even think of rdmx6-..., good catch! If we find more, then let's add them. Thanks!

**Comment by frederikrothenberger** [comment]
> Thanks Frederik! The text looks good! Just a small question - maybe we dont need the Logout button in this case?

Thanks. I removed the logout button.

File: src/frontend/src/flows/loginUnknown.ts (hunk 1 above)
Comment (frederikrothenberger) [reply src/frontend/src/flows/loginUnknown.ts:107]: Done.

**Review by nmattia — APPROVED** [review]
LGTM!

**Comment by frederikrothenberger** [comment]
Note: Not merging until 22.12.2021 due to deprecation period.

**Comment by frederikrothenberger** [comment]
Manual test of this feature on the identity test successful.

**Comment by ggreif** [comment]
Just a drive-by comment, does this code also apply to the testnets? We have https://rdmx6-jaaaa-aaaaa-aaadq-cai.peopleparty01.dfinity.network, will that be disabled too?

**Comment by frederikrothenberger** [comment]
@ggreif: No. This will only affect https://rdmx6-jaaaa-aaaaa-aaadq-cai.ic0.app and anything using .raw.

**Pull request merged**